Document that replacement functionality must be the same stability level (#42031)
kubernetes/website: content/en/docs/reference/using-api/deprecation-policy.md

   * **Beta: 3 months or 1 release (whichever is longer)**
   * **Alpha: 0 releases**

> **Rule #5c: CLI elements cannot be deprecated in favor of less stable
> CLI elements**

Reviewer: ```suggestion
**Rule #5c: Command line interface (CLI) elements cannot be deprecated in favor of
less stable CLI elements**
```

> Similar to the Rule #3 for APIs, if CLI element is being replaced with the

Reviewer: ```
suggestion
Similar to the Rule #3 for APIs, if an element of a command line interface is being replaced with an
```

> alternative implementation, e.g. renamed elements or use of a config flag

Reviewer: ```suggestion
alternative implementation, such as by renaming an existing element, or by switching to
use configuration sourced from a file 
```

> instead of command line argument, that recommended alternative must be of

Reviewer: ```suggestion
instead of a command line argument, that recommended alternative must be of
```

**Rule #7: Deprecated behaviors must function for no less than 1 year after their
announced deprecation.**

If the feature or behavior is being replaced with an alternative implementation
that requires work to adopt the change, there should be an effort to simplify
the transition whenever possible. If an alternative implementation is under

> Kubernetes organization control, the following rule apply:

Reviewer 2: ```suggestion
Kubernetes organization control, the following rules apply:
```

**Rule #8: The feature of behavior must not be deprecated in favor of an alternative
implementation that is less stable**

> For example, GA feature cannot be deprecated in favor of a Beta replacement.

Reviewer: ```suggestion
For example, a generally available feature cannot be deprecated in favor of a Beta
replacement.
```

> We should, however, encourage the transition to the alternative implementation
> even before it reached the same maturity level. This is particularly important

Reviewer: ```suggestion
The Kubernetes project does, however, encourage users to adopt and transitions to alternative
implementations even before they reach the same maturity level. This is particularly important
```

We [avoid using “we”](https://kubernetes.io/docs/contribute/style/style-guide/#avoid-using-we).

> even before it reached the same maturity level. Especially for the new use of
> a feature or to get an early feedback on the replacement.

Reviewer 2: ```suggestion
even before it reached the same maturity level. This is particularly important
for exploring new use cases of a feature or obtaining early feedback on the replacement.
```

Alternative implementations may sometimes be external tools or products,

> for example the feature may move from the kubelet to container runtime

Reviewer: ```suggestion
for example a feature may move from the kubelet to container runtime
```

> that is not under Kuberentes project control. In such cases, the rule cannot be

Reviewer: ```suggestion
that is not under Kubernetes project control. In such cases, the rule cannot be
```

applied, but there must be an effort to ensure that there is a transition path

> that does not compromise on components maturity level. In the example with

Reviewer: ```suggestion
that does not compromise on components' maturity levels. In the example with
```

> container runtimes, the effort may involve ensuring that popular container runtimes

Reviewer: ```suggestion
container runtimes, the effort may involve trying to ensure that popular container runtimes
```

> container runtimes, the effort may be to ensure that popular container runtimes
> has a same stability level versions implementing that replacement behavior.

Reviewer 2: ```suggestion
container runtimes, the effort may involve ensuring that popular container runtimes have
versions that offer the same level of stability while implementing that replacement behavior.
```
